lib/api/unittest/CPersistenceManagerTest.cc

```diff
@@ -287,8 +293,104 @@ class CTestFixture {
+        static const std::string JOB_ID{"job"};
+
+        std::string inputFilename{"testfiles/big_ascending.txt"};
+
+        // Open the input and output files
+        std::ifstream inputStrm{inputFilename};
+        BOOST_TEST_REQUIRE(inputStrm.is_open());
+
+        std::ofstream outputStrm{ml::core::COsFileFuncs::NULL_FILENAME};
+        BOOST_TEST_REQUIRE(outputStrm.is_open());
+
+        ml::model::CLimits limits;
+        ml::api::CFieldConfig fieldConfig;
+        BOOST_TEST_REQUIRE(fieldConfig.initFromFile(configFileName));
+
+        ml::model::CAnomalyDetectorModelConfig modelConfig{
+            ml::model::CAnomalyDetectorModelConfig::defaultConfig(BUCKET_SIZE)};
+
+        std::ostringstream* dataStream{nullptr};
+        ml::api::CSingleStreamDataAdder::TOStreamP dataStreamPtr{
+            dataStream = new std::ostringstream()};
+
+        // Persist the processors' state
+        ml::api::CSingleStreamDataAdder dataAdder{dataStreamPtr};
+
+        // The 30000 second persist interval is set large enough that the timer
+        // will not trigger during the test - we bypass the timer in this test
+        // and kick off the background persistence chain explicitly
+        ml::api::CPersistenceManager persistenceManager{30000, false, dataAdder};
+
+        std::string description_;
+        std::string snapshotId_;
+        std::size_t numDocs_{0};
+
+        std::string backgroundSnapshotId;
+        std::string foregroundSnapshotId;
+
+        {
+            ml::core::CJsonOutputStreamWrapper wrappedOutputStream{outputStrm};
+
+            CTestAnomalyJob job{JOB_ID,
+                                limits,
+                                fieldConfig,
+                                modelConfig,
+                                wrappedOutputStream,
+                                std::bind(&reportPersistComplete,
+                                          std::placeholders::_1, std::ref(description_),
+                                          std::ref(snapshotId_), std::ref(numDocs_)),
+                                &persistenceManager,
+                                -1,
+                                "time",
+                                "%d/%b/%Y:%T %z"};
+
+            ml::api::CDataProcessor* firstProcessor{&job};
+
+            ml::api::CNdJsonInputParser parser{
+                {CTestFieldDataCategorizer::MLCATEGORY_NAME}, inputStrm};
+
+            BOOST_TEST_REQUIRE(parser.readStreamIntoMaps(
+                [firstProcessor](const ml::api::CDataProcessor::TStrStrUMap& dataRowFields) {
+                    return firstProcessor->handleRecord(
+                        dataRowFields, ml::api::CDataProcessor::TOptionalTime{});
+                }));
+
+            // Ensure the model size stats are up to date
+            job.finalise();
+
+            ml::core_t::TTime snapshotTimestamp{1283524206};
+            const std::string snapshotId{ml::core::CStringUtils::typeToString(snapshotTimestamp)};
```

Comment:
It's probably best to set the snapshot ID to something different to the snapshot timestamp, just to prove that the code being tested doesn't mix up the snapshot ID and the snapshot timestamp.

```suggestion
            const std::string snapshotId{"my_special_snapshot"};
```